### PR TITLE
Decompose `Exp` into `PauliRot` if possible

### PR DESCRIPTION
**Pull request by albi3ro**
Right now, the `Exp` class doesn't have a decomposition.

This PR decomposes `Exp` into `PauliRot` if:
* The coefficient is purely imaginary
* The base operator is a pauli word

For example:

```pycon
>>> op = qml.exp(qml.PauliX(0) @ qml.PauliY(1), -0.5j * 1.234)
>>> op.decomposition()
[PauliRot(-1.234, XY, wires=[0, 1])]
```

It also updates the repr method for `PauliRot`, because it was missing crucial information.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -233,6 +233,9 @@ Users can specify the control wires as well as the values to control the operati
 
 <h3>Improvements</h3>
 
+* The `Exp` class decomposes into a `PauliRot` class if the coefficient is imaginary and 
+  the base operator is a Pauli Word.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:237]:
```suggestion
  the base operator is a Pauli Word.
  [(#3249)](https://github.com/PennyLaneAI/pennylane/pull/3249)
```

File: pennylane/ops/op_math/exp.py [hunk 2]
```diff
@@ -20,9 +20,11 @@
 import pennylane as qml
 from pennylane import math
 from pennylane.operation import (
+    DecompositionUndefinedError,
     expand_matrix,
     OperatorPropertyUndefined,
 )
+from pennylane.pauli import is_pauli_word, pauli_word_to_string
```
Comment (Jaybsoni) [inline pennylane/ops/op_math/exp.py:27]: <3

File: pennylane/ops/op_math/exp.py [hunk 3]
```diff
@@ -165,6 +167,18 @@ def _queue_category(self):
             return "_ops"
         return None
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_decomposition(self):
+        return math.real(self.coeff) == 0 and is_pauli_word(self.base)
+
+    def decomposition(self):
+        if math.real(self.coeff) != 0 or not is_pauli_word(self.base):
```
Comment (Jaybsoni) [inline pennylane/ops/op_math/exp.py:176]: Maybe add docstring?

**Review by Jaybsoni — APPROVED** [review]
Looks good to go! Just add some quick suggestions 💯

File: tests/ops/op_math/test_exp.py [hunk 4]
```diff
@@ -275,6 +276,42 @@ def test_sparse_matrix_wire_order_error(self):
...
+    def test_pauli_word_decompositioni(self, base, base_string):
+        """Check that Exp decomposes into PauliRot if possible."""
+        theta = 3.21
+        op = Exp(base, -0.5j * theta)
+
+        assert op.has_decomposition
+        pr = op.decomposition()[0]
+        assert pr.data[0] == 3.21
+        assert pr.hyperparameters["pauli_word"] == base_string
+        assert pr.wires == base.wires
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply tests/ops/op_math/test_exp.py:312]: Just realized I can use `qml.equal` 😆

File: pennylane/ops/op_math/exp.py (hunk 3 above)
Comment (albi3ro) [reply pennylane/ops/op_math/exp.py]: added

File: pennylane/ops/op_math/exp.py [hunk 5]
```diff
@@ -165,6 +167,29 @@ def _queue_category(self):
...
+        .. math:: O = O_1 O_2 \dots O_n
+
+        A ``DecompositionUndefinedError`` is raised if the coefficient is not imaginary or the base
+        is not a Pauli Word.
+
+        Returns:
+            list[PauliRot]: decomposition of the operator
+        """
+        if math.real(self.coeff) != 0 or not is_pauli_word(self.base):
+            raise DecompositionUndefinedError
```
Comment (trbromley) [inline pennylane/ops/op_math/exp.py:188]: Could we use `has_decomposition` here?

**Pull request merged**